src/main/java/step3/LottoNumber.java

```diff
@@ -0,0 +1,40 @@
+package step3;
+
+public class LottoNumber {
+    private static final int MAX_LOTTO_NUMBER = 45;
+    private static final int MIN_LOTTO_NUMBER = 1;
+    private final int number;
+    private boolean isBonus = false;
+
+    LottoNumber(int number) {
+        validateNumber(number);
+        this.number = number;
+    }
+
+    private void validateNumber(int number) {
+        if (number < MIN_LOTTO_NUMBER || MAX_LOTTO_NUMBER < number) {
+            throw new IllegalArgumentException();
+        }
+    }
+
+    public int getNumber() {
+        return number;
+    }
+
+    public boolean isBonus() {
+        return isBonus;
+    }
+
+    public void setBonus(boolean bonus) {
+        isBonus = bonus;
+    }
+
+    public boolean compareToNumber(LottoNumber other) {
+        return this.number == other.number;
```

Comment:
compare 메소드라면
```suggestion
    public boolean compareToNumber(LottoNumber other) {
        return Integer.compare(this.number, other.number);
```
가 적절해보입니다:)

아니면 `equals()`와 `hashCode()`를 재정의하셔도 좋을 것 같네요:)